### PR TITLE
Switiching to apps/v1 deployment api in the examples

### PR DESCRIPTION
**Pull request by yue9944882**
Fixes: #1075

### CODE REVIEW

File: examples/src/main/java/io/kubernetes/client/examples/PatchExample.java [hunk 1]
```diff
@@ -102,17 +101,17 @@ public static void main(String[] args) throws IOException {
       // apply-yaml a deployment, server side apply is alpha in kubernetes v1.14,
       // You need to actively enable the Server Side Apply alpha feature
       // https://kubernetes.io/docs/reference/using-api/api-concepts/#server-side-apply
```
Comment (php-coder) [inline examples/src/main/java/io/kubernetes/client/examples/PatchExample.java:103]: Is this comment is still needed given in the lite of 1.18?

File: examples/src/main/java/io/kubernetes/client/examples/PatchExample.java [hunk 2]
```diff
@@ -102,17 +101,17 @@ public static void main(String[] args) throws IOException {
...
-      ExtensionsV1beta1Deployment deploy4 =
+      V1Deployment deploy4 =
           PatchUtils.patch(
```
Comment (innulic) [inline examples/src/main/java/io/kubernetes/client/examples/PatchExample.java:105]: @yue9944882 I run your version of example and for yaml apply I received an error:

{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"failed to create typed patch object: .spec.template.spec.containers[name=\"hello-node\"].ports: element 0: associative list with keys has an element that omits key field \"protocol\"","code":500}

File: examples/src/main/java/io/kubernetes/client/examples/PatchExample.java (hunk 1 above)
Comment (yue9944882) [reply examples/src/main/java/io/kubernetes/client/examples/PatchExample.java:103]: good catch, i updated the comment to clarify that server-side-apply is enabled by default after v1.16

File: examples/src/main/java/io/kubernetes/client/examples/PatchExample.java (hunk 2 above)
Comment (yue9944882) [reply examples/src/main/java/io/kubernetes/client/examples/PatchExample.java:105]: can you retry on my latest update? i just re'produced the error on a v1.18 cluster and fixed it.

Comment (innulic) [reply examples/src/main/java/io/kubernetes/client/examples/PatchExample.java:105]: thank you, now it works

**Comment by brendandburns** [comment]
/lgtm

Thanks!

**Pull request merged**